vtex-apps/admin-catalog-translations
# Collections translation

**What problem is this solving?**


This PR allows user to translate collection names. Different than the other components in this application, the component uses `vtex.messages` app to get and translate the entries.

This PR also brings localization to some common components, like `react/components/ActionButtons.tsx` and `react/components/ErrorHandler.tsx`

The main component `react/components/CollectionsTranslation/CollectionsTranslation.tsx` uses the `useCatalogQuery` to fetch the collection name from `catalog-graphql`. Since it doesn't translate collections, it always return the collection name for the store main language. We leverage that information [to call](https://github.com/vtex-apps/admin-catalog-translations/compare/collections-translation?expand=1#diff-2e4d10a7ac54aeda2e0bdb06b5199fcec05949f20368368b3bfc80277b2cb532R40-R63) `userTranslations`from `messages` app. This query already brings all the translations available for a given language, a context (collection id) and a content (string to be translated from the main language). However, it only brings the translations already submitted by a user. There will be no information for the other languages. In this case, [we render the value for the main language](https://github.com/vtex-apps/admin-catalog-translations/compare/collections-translation?expand=1#diff-2e4d10a7ac54aeda2e0bdb06b5199fcec05949f20368368b3bfc80277b2cb532R75-R78).

In the `react/components/CollectionsTranslation/CollectionsForm.tsx`, the app [sends the translation to messages api](https://github.com/vtex-apps/admin-catalog-translations/compare/collections-translation?expand=1#diff-c58cb53d116e49f2b1c83703519e552af7d2a460fb13dd26b8ff209a97057ad1R34-R71). After sending the mutation, it calls `        updateMemoCollections` in order to re-render the component with the new translation.

The helper function [formatCollectionFromMessages](https://github.com/vtex-apps/admin-catalog-translations/compare/collections-translation?expand=1#diff-602159d3caf4faefc0229210cda914e9991d880722203cf64b6a373951862dadR142-R158) handles the cases where there was no translation yet for a collection. Since the `userTranslations` query returns an empty array until the first translation is sent, we fake the translation for the main language, using the srcLang as key and the entry value from catalog-graphql as value.

**How should this be manually tested?**

In [this workspace](https://catalogcollectionstranslate--powerplanet.myvtex.com/admin/catalog-translation/collections), search for any available collection - e.g. 141, 142, 143, 144, 146. You should see the original value for the collections. 

Navigate to another language and submit a new value. The entry will be updated if everything goes right.

**Screenshots or example usage:**

![recording (3)](https://user-images.githubusercontent.com/38737958/126342036-f78438f2-b9da-4cd7-b931-819ffe01be45.gif)
